robocluster/member.py

```diff
@@ -0,0 +1,374 @@
+import asyncio
+import socket
+import json
+import os
+from fnmatch import fnmatch
+from ipaddress import IPv4Network
+
+from .net import AsyncSocket
+from .looper import Looper
+from .util import as_coroutine
+
+
+class Error(Exception):
+    pass
+
+
+class UnknownPeer(Error):
+    pass
+
+
+class Member(Looper):
+    def __init__(self, name, network, port, key=None, loop=None):
+        super().__init__(loop)
+        self.name = name
+        self.uid = int.from_bytes(os.urandom(4), 'big')
+        self.wanted = set()
+
+        self.subscriptions = set()
+
+        self._send_endpoints = {}
+        self._request_endpoints = {}
+
+        self._peers = {}
+        self._accepter = _Accepter(self)
+        self._gossiper = _Gossiper(self, network, port, key=key)
+
+    def try_peer(self, peer):
+        try:
+            return self._peers[peer]
+        except KeyError:
+            raise UnknownPeer(peer)
+
+    def on_recv(self, endpoint, callback):
+        self._send_endpoints[endpoint] = as_coroutine(callback)
```

Comment:
Would it maybe be a good idea to check if a callback is already registered to avoid users accidentally overwriting callbacks?